code/modules/vehicles/mecha/equipment/weapons/weapons.dm

```diff
@@ -4,54 +4,217 @@
 	equipment_slot = MECHA_WEAPON
 	destroy_sound = 'sound/mecha/weapdestr.ogg'
 	mech_flags = EXOSUIT_MODULE_COMBAT
-	/// ammo datum typepath
+	/// ammo datum or object typepath
 	var/ammotype
-	///sound file to play when this weapon does it's action
+	///sound file to play when this weapon you know, fires
 	var/fire_sound
-	/// how many projectiles this weapon
-	var/projectiles_per_shot = 1
-	/// basically spread
+	///current tracked target for fire(), updated when user drags
+	var/atom/current_target
+	///current mob firing this weapon. used for tracking for iff and etc in fire()
+	var/mob/current_firer
+	///Tracks windups
+	var/windup_checked = WEAPON_WINDUP_NOT_CHECKED
+	///Muzzle flash visual reference
+	var/atom/movable/vis_obj/effect/muzzle_flash/muzzle_flash
+	///list for this weapons flash offsets Format: MECHA_SLOT = list(DIR = list(PIXEL_X, PIXEL_Y))
+	var/list/flash_offsets = list(
+		MECHA_R_ARM = list("N" = list(0,0), "S" = list(0,0), "E" = list(0,0), "W" = list(0,0)),
+		MECHA_L_ARM = list("N" = list(0,0), "S" = list(0,0), "E" = list(0,0), "W" = list(0,0)),
+	)
+	///Icon state of the muzzle flash effect.
+	var/muzzle_iconstate
+	///color of the muzzle flash while shooting
+	var/muzzle_flash_color = COLOR_VERY_SOFT_YELLOW
+	///Range of light when we flash while shooting
+	var/muzzle_flash_lum = 3
+	///windup sound played during windup
+	var/windup_sound
+	///windup delay for this object
+	var/windup_delay = 0
+	///scatter of this weapon. in degrees and modified by arm this is attached to
 	var/variance = 0
```

Comment:
then why is it called variance btw ?

Reply:
Uhhh because it didn't want to remove the tg var to make any later improvements less annoying to port but I can remove